### PR TITLE
#CX-11353 : agent desktop causes memory spike on local PC

### PR DESCRIPTION
**Pull request by yasagraw**
## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/date-range-picker/DateRangePicker.ts [hunk 1]
```diff
@@ -46,7 +44,7 @@ export namespace DateRangePicker {
       return date.replace(/-+/g, "/");
     }
 
-    handleDateSelection = (e: CustomEvent) => {
+    handleDateSelection(e: any){
```
Comment (parsoman) [inline web-components/src/components/date-range-picker/DateRangePicker.ts:47]: This should be a arrow function.

File: web-components/src/components/taskitem/TaskItem.ts [hunk 2]
```diff
@@ -199,8 +199,8 @@ export namespace TaskItem {
           class="md-taskitem ${classMap({ selected: this.selected })}"
           tabindex="0"
           aria-selected="${this.selected}"
-          @click=${(e: MouseEvent) => this.handleClick(e)}
-          @keydown=${(e: KeyboardEvent) => this.handleKeyDown(e)}
+          @click=${this.handleClick}
```
Comment (parsoman) [inline web-components/src/components/taskitem/TaskItem.ts:202]: I dont think this changes have any impact to performance. Please revert this kind of changes. @prakhar-tiwari Right?

Comment (yasagraw) [reply web-components/src/components/taskitem/TaskItem.ts:202]: I think it is better to not use arrow function here. As using arrow function will separate function whenever it gets called

File: web-components/src/components/date-range-picker/DateRangePicker.ts (hunk 1 above)
Comment (yasagraw) [reply web-components/src/components/date-range-picker/DateRangePicker.ts]: Done

File: web-components/src/components/dropdown/Dropdown.ts [hunk 3]
```diff
@@ -481,11 +487,7 @@ export namespace Dropdown {
                   aria-selected="${idx === this.focusedIndex}"
                   part="dropdown-option"
                   ?focused="${idx === this.focusedIndex}"
-                  @click="${() => {
-                    this.focusToIndex(idx);
-                    this.select();
-                    this.collapse();
-                  }}"
+                  @click="${() => this.onListItemClick(idx)}"
```
Comment (Samreen26) [inline web-components/src/components/dropdown/Dropdown.ts:490]: here also can we change it to `@click="${this.onListItemClick(idx)}"` ?

File: web-components/src/components/taskitem/TaskItem.ts (hunk 2 above)
Comment (yasagraw) [reply web-components/src/components/taskitem/TaskItem.ts]: Removed these changes

**Review by prakhar-tiwari — APPROVED** [review]

**Review by momeraj — APPROVED** [review]

File: web-components/src/components/datepicker/datepicker-day/DatePickerDay.ts [hunk 4]
```diff
@@ -59,16 +59,18 @@ export namespace DatePickerDay {
...
-        new CustomEvent("day-select", {
-          bubbles: true,
-          composed: true,
-          detail: {
-            date: this.day,
-            sourceEvent: e
-          }
-        })
-      );
+      if (!this.disabled) {
```
Comment (parsoman) [inline web-components/src/components/datepicker/datepicker-day/DatePickerDay.ts:62]: THis new check is required, was it a bug earlier?

**Review by parsoman — DISMISSED** [review]

File: web-components/src/components/datepicker/datepicker-day/DatePickerDay.ts (hunk 4 above)
Comment (yasagraw) [reply web-components/src/components/datepicker/datepicker-day/DatePickerDay.ts]: Removed this check. Not required here

**Review by prakhar-tiwari — DISMISSED** [review]

**Review by parsoman — APPROVED** [review]

**Review by parsoman — APPROVED** [review]

**Pull request merged**